### PR TITLE
feat: add EC component contextual-navigation - INNO-1352

### PR DESCRIPTION
**Pull request by kalinchernev**
# PR description

I have added an option to set number of items to display. It's not the case comparing with React component, though I have set a default, which matches React component, given there are is no input for that from the user.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [x] I have put the vanilla component as `devDependencies`
- [x] I have put the specs package as `devDependencies`
- [x] I have added the components directly used in the twig file (with `include` or `embed`) as `dependencies`
- [x] My component is listed in `@ecl-twig/ec-components`'s `dependencies`
- [x] My variables naming follow the guidelines (snake case for twig)
- [x] I have provided tests
- [x] I have provided documentation (for the "notes" tab)
- [x] If my local `yarn.lock` contains changes, I have committed it
- [x] I have given my PR the proper label (`pr: review needed` to indicate that I'm done and now waiting for a review ,`pr: wip` to indicate that I'm actively working on it ...)

### CODE REVIEW

File: src/ec/packages/ec-component-contextual-navigation/contextual-navigation.html.twig [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result #}
+
+<nav
+  class="{{ _css_class }}"{{ _extra_attributes|raw }}
+  data-ecl-contextual-navigation>
```
Comment (yhuard) [inline src/ec/packages/ec-component-contextual-navigation/contextual-navigation.html.twig]: it's more readable with a newline before the closing `>` :)

File: src/ec/packages/ec-component-contextual-navigation/contextual-navigation.html.twig [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  <div class="ecl-contextual-navigation__label">{{- _label -}}</div>
```
Comment (yhuard) [inline src/ec/packages/ec-component-contextual-navigation/contextual-navigation.html.twig]: no need to remove whitespaces here

File: src/ec/packages/ec-component-contextual-navigation/contextual-navigation.html.twig [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    data-ecl-contextual-navigation-list
+  >
+    {% if _items is not empty and _items is iterable %}
+      {% for item in _items %}
+        {% set _item_classes = 'ecl-contextual-navigation__item' %}
+        {% if loop.index > _items_limit %}
+          {% set _item_classes = _item_classes ~ ' ' ~ 'ecl-contextual-navigation__item--collapsed' %}
+        {% endif %}
+        <li class="{{ _item_classes }}">
+        {%- include '../ec-component-link/link.html.twig' with item|merge({
```
Comment (yhuard) [inline src/ec/packages/ec-component-contextual-navigation/contextual-navigation.html.twig]: missing indent + unnecessary whitespace control modifier

File: src/ec/packages/ec-component-contextual-navigation/contextual-navigation.html.twig [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+          link: item,
+        }) only -%}
+        </li>
+        {% if _item_more is not empty and loop.last %}
+          {% set _item_classes = _item_classes|replace({
+            'ecl-contextual-navigation__item--collapsed': '',
+          }) %}
+          {% set _item_classes = _item_classes ~ ' ' ~ 'ecl-contextual-navigation__item--more' %}
+          <li class="{{ _item_classes }}">
+          {%- include '../ec-component-button/button.html.twig' with _item_more|merge({
```
Comment (yhuard) [inline src/ec/packages/ec-component-contextual-navigation/contextual-navigation.html.twig]: missing indent + unnecessary whitespace control modifier

File: src/ec/packages/ec-component-contextual-navigation/__snapshots__/contextual-navigation.test.js.snap [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+        Item 6
+      </a>
+    </li>
+    <li
+      class="ecl-contextual-navigation__item  ecl-contextual-navigation__item--more"
+    >
+      <button
+        class="ecl-button ecl-button--ghost ecl-contextual-navigation__more"
+        data-ecl-contextual-navigation-more="true"
+        type="submit"
```
Comment (yhuard) [inline src/ec/packages/ec-component-contextual-navigation/__snapshots__/contextual-navigation.test.js.snap]: it shouldn't be a `submit` button ;) (`submit` is the default type)

Comment (kalinchernev) [reply src/ec/packages/ec-component-contextual-navigation/__snapshots__/contextual-navigation.test.js.snap]: I'm not sure which one it should be, I followed the [current one](https://v2-dev--europa-component-library.netlify.com/ec/components/navigation/contextual-navigation/code/)
